### PR TITLE
feat(forms): add async headers

### LINKED ISSUE #346: Async headers for subscriber's form pages
Add two async headers for subscriber's form pages, i.e. first header shows a form's name and second header show a form's name, button to activate/deactivate form and a dropdown with actions for deleting, copy form's url and copy form's code.

image

### PR DESCRIPTION
**Pull request by zouxuoz**
resolve #346

### CODE REVIEW

**Comment by astupak** [comment]
Add forms API module description to docs

File: packages/sendsay-api/modules/forms/parsers.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+};
+
+export const getFormDraft = res => ({
+  name: res.obj.data.name,
+  status: 'draft',
+});
+
+export const getForm = res => ({
+  name: res.obj.name,
+  active: !!+res.obj.state,
```
Comment (rosendi) [inline packages/sendsay-api/modules/forms/parsers.js]: Please, use `parseInt`.

File: packages/sendsay-api/modules/forms/parsers.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+export const getFormFields = res => res.obj.order.reduce((result, fieldId) => {
+  if (['-confirmed', '-dt', '-merge_fields', '-merge_answers'].indexOf(fieldId) === -1) {
+    const field = res.obj.quests[fieldId];
+
+    const temp = {
+      id: fieldId,
+      type: typesMap[field.type],
+      name: field.name,
+      defaultValue: field.form.default || null,
+      hidden: !!+field.form.hidden,
```
Comment (rosendi) [inline packages/sendsay-api/modules/forms/parsers.js]: Please, use `parseInt`.

File: packages/sendsay-api/modules/forms/parsers.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      required: !!+field.form.mustbe,
```
Comment (rosendi) [inline packages/sendsay-api/modules/forms/parsers.js]: Please, use `parseInt`.

File: src/routes/subscribers/routes/forms/routes/show/components/FormLayout.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    <FormHeaderContainer formId={formId} form={form} />
+    {form.status !== 'draft' && <FormNavigation intl={intl} formId={formId} />}
+    {children}
+  </div>
+);
+
+FormLayout.propTypes = {
+  intl: intlShape.isRequired,
+  formId: PropTypes.string.isRequired,
+  form: PropTypes.shape({
```
Comment (rosendi) [inline src/routes/subscribers/routes/forms/routes/show/components/FormLayout.js]: Put it after `children`

File: packages/sendsay-legacy/kit/page/subscribers/forms/index.js [hunk 5]
```diff
@@ -15,7 +16,7 @@ define(function(require) {
                 var page = this;
 
                 return new FormModel({
-                    id: page.params.id
+                    id: (page.params || {}).id
```
Comment (radiansz) [inline packages/sendsay-legacy/kit/page/subscribers/forms/index.js]: Please, &&

File: src/routes/subscribers/routes/forms/routes/show/components/FormLayout.js (hunk 4 above)
Comment (rosendi) [inline src/routes/subscribers/routes/forms/routes/show/components/FormLayout.js]: With a new line.

**Comment by radiansz** [comment]
lgtm

**Comment by rosendi** [comment]
lgtm

**Pull request merged**